src/uct/ugni/rdma/ugni_rdma_iface.c

```diff
@@ -0,0 +1,272 @@
+    .ep_destroy          = UCS_CLASS_DELETE_FUNC_NAME(uct_ugni_ep_t),
+    .ep_put_short        = uct_ugni_ep_put_short,
+    .ep_put_bcopy        = uct_ugni_ep_put_bcopy,
+    .ep_put_zcopy        = uct_ugni_ep_put_zcopy,
+    .ep_am_short         = uct_ugni_ep_am_short,
+    .ep_atomic_add64     = uct_ugni_ep_atomic_add64,
+    .ep_atomic_fadd64    = uct_ugni_ep_atomic_fadd64,
+    .ep_atomic_cswap64   = uct_ugni_ep_atomic_cswap64,
+    .ep_get_bcopy        = uct_ugni_ep_get_bcopy,
+    .ep_get_zcopy        = uct_ugni_ep_get_zcopy,
+};
+
+static UCS_CLASS_INIT_FUNC(uct_ugni_rdma_iface_t, uct_pd_h pd, uct_worker_h worker,
+                           const char *dev_name, size_t rx_headroom,
+                           const uct_iface_config_t *tl_config)
+{
+    uct_ugni_iface_config_t *config = ucs_derived_of(tl_config, uct_ugni_iface_config_t);
+    ucs_status_t rc;
+
+    pthread_mutex_lock(&uct_ugni_global_lock);
+
+    UCS_CLASS_CALL_SUPER_INIT(uct_ugni_iface_t, pd, worker, dev_name, &uct_ugni_iface_ops,
+                              &config->super UCS_STATS_ARG(NULL));
+
+    /* Setting initial configuration */
+    self->config.fma_seg_size  = UCT_UGNI_MAX_FMA;
+    self->config.rdma_max_size = UCT_UGNI_MAX_RDMA;
+
+    rc = ucs_mpool_create("UGNI-DESC-ONLY", sizeof(uct_ugni_base_desc_t),
+                          0,                            /* alignment offset */
+                          UCS_SYS_CACHE_LINE_SIZE,      /* alignment */
+                          128 ,                         /* grow */
+                          config->mpool.max_bufs,       /* max buffers */
+                          &self->super.super,           /* iface */
+                          ucs_mpool_hugetlb_malloc,     /* allocation hooks */
+                          ucs_mpool_hugetlb_free,       /* free hook */
+                          uct_ugni_base_desc_init,      /* init func */
+                          NULL , &self->free_desc);
+    if (UCS_OK != rc) {
+        ucs_error("Mpool creation failed");
+        goto exit;
+    }
+
+    rc = ucs_mpool_create("UGNI-GET-DESC-ONLY", sizeof(uct_ugni_fetch_desc_t),
+                          0,                            /* alignment offset */
+                          UCS_SYS_CACHE_LINE_SIZE,      /* alignment */
+                          128 ,                         /* grow */
+                          config->mpool.max_bufs,       /* max buffers */
+                          &self->super.super,           /* iface */
+                          ucs_mpool_hugetlb_malloc,     /* allocation hooks */
+                          ucs_mpool_hugetlb_free,       /* free hook */
+                          uct_ugni_base_desc_init,      /* init func */
+                          NULL , &self->free_desc_get);
+    if (UCS_OK != rc) {
+        ucs_error("Mpool creation failed");
+        goto clean_desc;
+    }
+
+    rc = ucs_mpool_create("UGNI-DESC-BUFFER",
```

Comment:
Not directly related to this change, but I want to raise it here. Can we replace ucs_mpool_create() with uct_iface_mpool_create()